src/WebJobs.Script/Diagnostics/CategoryTraceWriterFactory.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    internal class CategoryTraceWriterFactory
+    {
+        private readonly string _category;
+        private readonly ScriptHostConfiguration _scriptHostConfig;
+
+        public CategoryTraceWriterFactory(string category, ScriptHostConfiguration scriptHostConfig)
```

Comment:
Is there another use case for a new log file? I guess my main concern is that right now if something is busted you need to look in Host logs and Function logs. This introduces a new file to look at -- and others will increase that list.

I'm not completely opposed -- as I said, sending everything to App Insights will be more useful -- I just want to make sure we're not complicating anything.